owner/src/main/java/org/the_chance/honeymart/ui/features/signup/SignUpViewModel.kt

```diff
@@ -21,19 +23,20 @@ class SignUpViewModel @Inject constructor(
     private val createMarketUseCase: CreateMarketUseCase,
     private val addMarketImageUseCase: AddMarketImageUseCase,
     private val validateMarketFieldsUseCase: ValidateMarketFieldsUseCase,
-    private val loginOwnerUseCase: LoginOwnerUseCase
+    private val loginOwnerUseCase: LoginOwnerUseCase,
+    private val stringResourceImpl: StringResource
```

Comment:
Naming would be better if it's `stringResource` instead of `stringResourceImpl` and as I mentioned consider changer the name to `dictionary`.